### PR TITLE
RL Training pipeline on 5-min data

### PR DESCRIPTION
**Pull request by lihuoran**
## How Has This Been Tested?

- [ ] Pass the test by running: `pytest qlib/tests/test_all_pipeline.py` under upper directory of `qlib`.
- [ ] If you are adding a new feature, test on your own test scripts.

## Screenshots of Test Results (if appropriate):
1. Pipeline test:
2. Your own tests:

## Types of changes

- [ ] Fix bugs
- [ ] Add new feature
- [ ] Update documentation

### CODE REVIEW

File: qlib/rl/order_execution/reward.py [hunk 1]
```diff
@@ -21,10 +21,13 @@ class PAPenaltyReward(Reward[SAOEState]):
     ----------
     penalty
         The penalty for large volume in a short time.
+    zoom
```
Comment (ultmaster) [inline qlib/rl/order_execution/reward.py:24]: Suggest "scale"

File: qlib/rl/contrib/train_onpolicy.py [hunk 2]
```diff
@@ -101,37 +103,47 @@ def train_and_test(
...
+            order_file_path=order_root_path / tag,
+            data_dir=Path(data_config["source"]["data_dir"]),
+            default_start_time_index=data_config["source"]["default_start_time"] // data_granularity,
+            default_end_time_index=data_config["source"]["default_end_time"] // data_granularity,
+        )
+        for tag in ("train", "valid", "test")
+    ]
 
-    callbacks = []
+    callbacks: List[Callback] = [MetricsWriter(dirpath=Path(trainer_config["checkpoint_path"]))]
```
Comment (ultmaster) [inline qlib/rl/contrib/train_onpolicy.py:131]: Putting this in the next-if as checkpoint_path is optional.

File: qlib/rl/data/pickle_styled.py [hunk 3]
```diff
@@ -100,6 +100,9 @@ def __init__(
         super(SimpleIntradayBacktestData, self).__init__()
 
         backtest = _read_pickle((data_dir if isinstance(data_dir, Path) else Path(data_dir)) / stock_id)
+        backtest = backtest.reset_index()
```
Comment (you-n-g) [inline qlib/rl/data/pickle_styled.py:103]: Will it be better to integrate these index operations into  `_read_pickle`?

File: qlib/rl/order_execution/reward.py (hunk 1 above)
Comment (you-n-g) [reply qlib/rl/order_execution/reward.py:24]: @lihuoran  Why is `zoom`/ `scale` necessary if we have hyperparameters like learning rate?

File: qlib/rl/trainer/vessel.py [hunk 4]
```diff
@@ -180,6 +180,7 @@ def train(self, vector_env: FiniteVectorEnv) -> Dict[str, Any]:
             update_result = self.policy.update(sample_size=0, buffer=collector.buffer, **self.update_kwargs)
             res = {**col_result, **update_result}
             self.log_dict(res)
+            del collector
```
Comment (you-n-g) [inline qlib/rl/trainer/vessel.py:183]: Doesn't Python delete the reference to the object automatically when the function is over?

File: qlib/rl/trainer/trainer.py [hunk 5]
```diff
@@ -218,6 +223,7 @@ def fit(self, vessel: TrainingVesselBase, ckpt_path: Path | None = None) -> None
             with _wrap_context(vessel.train_seed_iterator()) as iterator:
                 vector_env = self.venv_from_iterator(iterator)
                 self.vessel.train(vector_env)
+                del vector_env
```
[the remark this answers is not in the document]
Comment (lihuoran) [reply qlib/rl/trainer/trainer.py:226]: According to my experiments, memory will not be normally released after each training round without these explicit `del` operation, and that will cause OOM finally. I am not 100% sure about the mechanism here, but the `del` does work. Any better ideas?
CC @you-n-g

File: qlib/rl/trainer/trainer.py [hunk 6]
```diff
@@ -206,6 +207,10 @@ def fit(self, vessel: TrainingVesselBase, ckpt_path: Path | None = None) -> None
         self._call_callback_hooks("on_fit_start")
 
         while not self.should_stop:
+            msg = f"\n{datetime.now()}\tTrain iteration {self.current_iter + 1}/{self.max_iters}"
+            print(msg)
```
[the remark this answers is not in the document]
Comment (lihuoran) [reply qlib/rl/trainer/trainer.py:211]: Sorry, this is for test purpose only and I forget to remove it. A related question: can we show this message on screen immediately, in order to inform users about the training progress?

File: qlib/rl/order_execution/reward.py (hunk 1 above)
Comment (lihuoran) [reply qlib/rl/order_execution/reward.py:24]: Will be renamed to "scale".
@you-n-g In the AAAI 2021's open-source project, they use a scaled reward. I personally think adding this parameter would be more flexible.

File: qlib/rl/contrib/train_onpolicy.py [hunk 7]
```diff
@@ -160,6 +172,18 @@ def _simulator_factory_simple(order: Order) -> SingleAssetOrderExecutionSimple:
         vessel_kwargs=vessel_kwargs,
     )
 
+    backtest(
```
[the remark this answers is not in the document]
Comment (lihuoran) [reply qlib/rl/contrib/train_onpolicy.py:175]: Sure.

File: qlib/rl/trainer/vessel.py (hunk 4 above)
Comment (lihuoran) [reply qlib/rl/trainer/vessel.py:183]: Please see discussions above.

**Comment by you-n-g** [comment]
@lihuoran  Could you please check the CI errors?

File: qlib/rl/trainer/trainer.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (chenditc) [reply qlib/rl/trainer/trainer.py]: I did some testing on this. Memory and subprocess leak only reproducible when using GPU.
